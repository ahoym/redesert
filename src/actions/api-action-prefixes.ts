export interface ApiActionPrefixes {
  readonly FETCH: 'FETCH_';
  readonly CREATE: 'CREATE_';
  readonly UPDATE: 'UPDATE_';
  readonly REMOVE: 'REMOVE_';
}

const API_ACTION_PREFIXES: ApiActionPrefixes = {
  FETCH: 'FETCH_',
  CREATE: 'CREATE_',
  UPDATE: 'UPDATE_',
  REMOVE: 'REMOVE_',
};

export function resourceApiActionTypesFactory(resource: string) {
  return {
    FETCH: `${API_ACTION_PREFIXES.FETCH}_${resource}`,
    CREATE: `${API_ACTION_PREFIXES.CREATE}_${resource}`,
    UPDATE: `${API_ACTION_PREFIXES.UPDATE}_${resource}`,
    REMOVE: `${API_ACTION_PREFIXES.REMOVE}_${resource}`,
  };
}

export default API_ACTION_PREFIXES;
