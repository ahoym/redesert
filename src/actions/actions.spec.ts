import API_LIFECYCLE_SUFFIXES from './api-lifecycle-suffixes';
import isValidActionType from './is-valid-action-type';
import { resourceApiActionTypesFactory } from './api-action-prefixes';

describe('isValidActionType()', () => {
  const reducerNameSpace = 'FETCH_';

  it("is false for actionTypes that don't start with the reducerNamespace", () => {
    const actionType = `CREATE_foo_${API_LIFECYCLE_SUFFIXES.START}`;
    expect(isValidActionType(reducerNameSpace, actionType)).toEqual(false);
  });

  it("is false for actionTypes that don't end in one of the API_LIFECYCLE_SUFFIXES", () => {
    const actionType = 'CREATE_foo_BLARGH';
    expect(isValidActionType(reducerNameSpace, actionType)).toEqual(false);
  });

  it('is true only if the actionType has a valid namespace and suffix', () => {
    Object.keys(API_LIFECYCLE_SUFFIXES).forEach(suffix => {
      const actionType = `${reducerNameSpace}_foo_${suffix}`;
      expect(isValidActionType(reducerNameSpace, actionType)).toEqual(true);
    });
  });
});

describe('resourceApiActionTypesFactory()', () => {
  it('creates baseApiActionTypes for a passed in resource', () => {
    expect(resourceApiActionTypesFactory('foo')).toMatchSnapshot();
  });
});
