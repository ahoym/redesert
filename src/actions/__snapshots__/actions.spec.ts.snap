// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resourceApiActionTypesFactory() creates baseApiActionTypes for a passed in resource 1`] = `
Object {
  "CREATE": "CREATE__foo",
  "FETCH": "FETCH__foo",
  "REMOVE": "REMOVE__foo",
  "UPDATE": "UPDATE__foo",
}
`;
